container-search/src/main/java/com/yahoo/search/rendering/JsonGeneratorDataSink.java

```diff
@@ -0,0 +1,124 @@
+// Copyright Vespa.ai. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package com.yahoo.search.rendering;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.yahoo.data.disclosure.DataSink;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.charset.StandardCharsets;
+
+/**
+ * A sink that generates JSON from objects that implement {@link com.yahoo.data.disclosure.DataSource}.
+ *
+ * @author johsol
+ */
+record JsonGeneratorDataSink(JsonGenerator gen) implements DataSink {
+
+    @Override
+    public void fieldName(String utf16, byte[] utf8) {
+        try {
+            if (utf8 != null) {
+                gen.writeFieldName(new String(utf8, StandardCharsets.UTF_8));
+            } else {
+                gen.writeFieldName(utf16);
+            }
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    @Override
+    public void startObject() {
+        try {
+            gen.writeStartObject();
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    @Override
+    public void endObject() {
+        try {
+            gen.writeEndObject();
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    @Override
+    public void startArray() {
+        try {
+            gen.writeStartArray();
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    @Override
+    public void endArray() {
+        try {
+            gen.writeEndArray();
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    @Override
+    public void emptyValue() {
+        try {
+            gen.writeNull();
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    @Override
+    public void booleanValue(boolean v) {
+        try {
+            gen.writeBoolean(v);
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    @Override
+    public void longValue(long v) {
+        try {
+            gen.writeNumber(v);
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
```

Comment:
consider adding the overrides for all the data types that the generator supports